utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! `Structopt`-ready struct for `try-runtime`.
+
+use parity_scale_codec::{Decode, Encode};
+use std::{fmt::Debug, str::FromStr};
+use sc_service::Configuration;
+use sc_cli::{CliConfiguration, ExecutionStrategy};
+use sc_executor::{WasmExecutionMethod, NativeExecutor};
+use sp_state_machine::StateMachine;
+use sc_service::NativeExecutionDispatch;
+use sp_runtime::traits::{Block as BlockT, NumberFor};
+use sp_core::storage::{StorageData, StorageKey, well_known_keys};
+use frame_try_runtime::Target;
+
+/// Various commands to try out the new runtime, over configurable states.
+///
+/// For now this only assumes running the `on_runtime_upgrade` hooks.
+#[derive(Debug, structopt::StructOpt)]
+pub struct TryRuntimeCmd {
+	/// The shared parameters
+	#[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: sc_cli::SharedParams,
+
+	/// The target pallet to run the migration against.
+	#[structopt(short, long, default_value = "All")]
+	pub target: Target,
+
+	/// The state to use to run the migration. Should be a valid FILE or HTTP URI.
+	#[structopt(short, long, default_value = "http://localhost:9933")]
+	pub state: State,
+}
+
+/// The state to use for a migration dry-run.
+#[derive(Debug)]
+pub enum State {
+	/// A snapshot. Inner value is a file path.
+	Snap(String),
+
+	/// A live chain. Inner value is the HTTP uri.
+	Live(String),
+}
+
+impl FromStr for State {
+	type Err = &'static str;
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		match s.get(..7) {
+			// could use Url crate as well, but lets keep it simple for now.
+			Some("http://") => Ok(State::Live(s.to_string())),
+			Some("file://") => s
+				.split("//")
+				.collect::<Vec<_>>()
+				.get(1)
+				.map(|s| State::Snap(s.to_string()))
+				.ok_or("invalid file URI"),
+			_ => Err("invalid format. Must be a valid HTTP or File URI"),
+		}
+	}
+}
+
+impl TryRuntimeCmd {
+	pub async fn run<B, ExecDispatch>(&self, config: Configuration) -> sc_cli::Result<()>
+	where
+		B: BlockT,
+		ExecDispatch: NativeExecutionDispatch + 'static,
+	{
+
+		let spec = config.chain_spec;
+		let genesis_storage = spec.build_storage()?;
+
+		let code = StorageData(genesis_storage.top.get(well_known_keys::CODE).unwrap().to_vec());
+		let code_key = StorageKey(well_known_keys::CODE.to_vec());
+
+		let wasm_method = WasmExecutionMethod::Interpreted;
+		let strategy = ExecutionStrategy::Native;
+		let mut changes = Default::default();
+
+		let heap_pages = Some(1024);
+		let executor = NativeExecutor::<ExecDispatch>::new(wasm_method, heap_pages, 2);
```

Comment:
should probably end up being CLI flag configurable?

Reply:
Unless if you object, I would consider these only if/when they are needed, not now. For now, I prefer to keep this PR as lean as possible. Moreover, I suppose we would always prefer native execution in this use case.

Reply:
Oh well, but now I see that many of them are in the share params anyhow.. will fix.

Reply:
No -- they are not part of the default config, but not the shared params. Will add then.